static/app/components/metrics/queryBuilder.tsx

```diff
@@ -216,13 +216,45 @@ export const QueryBuilder = memo(function QueryBuilder({
   const projectIdStrings = useMemo(() => projectIds.map(String), [projectIds]);
 
   return (
-    <QueryBuilderWrapper>
+    <QueryBuilderWrapper hasMetricsNewInputs={hasMetricsNewInputs(organization)}>
+      {hasMetricsNewInputs(organization) && (
+        <GuideAnchor target="metrics_selector" position="bottom" disabled={index !== 0}>
+          <QueryFieldGroup>
+            <QueryFieldGroup.Label>{t('Visualize')}</QueryFieldGroup.Label>
+            <MRISelect
+              onChange={handleMRIChange}
+              onTagClick={handleMetricTagClick}
+              onOpenMenu={handleOpenMetricsMenu}
+              isLoading={isMetaLoading}
+              metricsMeta={meta}
+              projects={projectIds}
+              value={metricsQuery.mri}
+            />
+          </QueryFieldGroup>
+        </GuideAnchor>
+      )}
       <FlexBlock>
-        <FlexBlock>
-          <GuideAnchor target="metrics_selector" position="bottom" disabled={index !== 0}>
-            {hasMetricsNewInputs(organization) ? (
-              <QueryFieldGroup>
-                <QueryFieldGroup.Label>{t('Visualize')}</QueryFieldGroup.Label>
+        {!hasMetricsNewInputs(organization) && (
+          <FlexBlock>
+            <GuideAnchor
+              target="metrics_selector"
+              position="bottom"
+              disabled={index !== 0}
+            >
+              {hasMetricsNewInputs(organization) ? (
```

Comment:
This check is redundant, it will always be false, because of the check on line 237

Reply:
oh yes! good catch